### PR TITLE
Add testcases to validate consistency of packDataNsec

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

### CODE REVIEW

File: msg_helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+					0, 19, 48, 48,
+				},
+				off: 48,
+			},
+			wantErr:    true,
+			wantErrMsg: "dns: overflow packing nsec",
+			want:       31,
+		},
+		{
+			name: "disordered nseec bits",
```
Comment (miekg) [inline msg_helpers_test.go]: nsec

File: msg_helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package dns
+
+import "testing"
+
+func TestPackDataNsec(t *testing.T) {
```
Comment (miekg) [inline msg_helpers_test.go]: Can you add some comments on how you generated these test cases?

Comment (fntlnz) [reply msg_helpers_test.go]: added such comment

File: msg_helpers_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
+package dns
+
+import "testing"
+
+// TestPacketDataNsec tests generated using fuzz.go and with a message pack
+// containing the following bytes: 0000\x00\x00000000\x00\x002000000\x0060000\x00\x130000000000000000000"
+// That bytes sequence created the overflow error and further permutations of that sequence were able to trigger
+// the other code paths.
```
Comment (miekg) [inline msg_helpers_test.go]: uhm, shouldn't those fixes part of this PR?

Comment (fntlnz) [reply msg_helpers_test.go]: Can you elaborate more ?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Add testcases to va..." ]
>fntlnz commented on this pull request.
>
>
>
>> @@ -0,0 +1,108 @@
>+package dns
>+
>+import "testing"
>+
>+// TestPacketDataNsec tests generated using fuzz.go and with a message pack
>+// containing the following bytes: 0000\x00\x00000000\x00\x002000000\x0060000\x00\x130000000000000000000"
>+// That bytes sequence created the overflow error and further permutations of that sequence were able to trigger
>+// the other code paths.
>
>Can you elaborate more ?

you say you saw a crash? If so we should apply a fix?

**Comment by fntlnz** [comment]
I only got a defined error path while passing that input, not a panic or anything worse.

**Pull request merged**